components/admin_console/group_settings/group_details/group_details.jsx

```diff
@@ -137,10 +137,16 @@ export default class GroupDetails extends React.PureComponent {
         const {group} = this.props;
         const originalAllowReference = group.allow_reference;
         const saveNeeded = true;
+        let {groupMentionName} = this.state;
+
+        if (!originalAllowReference && allowReference && (typeof groupMentionName === 'undefined' || groupMentionName === '')) {
```

Comment:
I think you could just do 
```suggestion
        if (!originalAllowReference && allowReference && !groupMentionName) {
```